### ISSUE TITLE
[Accessibility] NVDA does not read text correctly PDF text in pdf.js

### ISSUE THREAD

**Issue by antoniolucasnobar** [mozilla/pdf.js#18297]
Attach (recommended) or Link to PDF file here:

Configuration:
- Web browser and its version: Firefox (127.0.1) and MS Edge (Chromium) - 126.0.2592.56
- Operating system and its version: Windows 11
- PDF.js version: 4.4.97 (https://mozilla.github.io/pdf.js/web/viewer.html)
- Is a browser extension: No

Steps to reproduce the problem:
1. Open one (or both) of these 2 PDFs in https://mozilla.github.io/pdf.js/web/viewer.html 
[Documento_984d1b9 (1).pdf](https://github.com/user-attachments/files/15903951/Documento_984d1b9.1.pdf) - this one is generated using Apache PDFBox
[Documento_984d1b9.pdf](https://github.com/user-attachments/files/15903859/Documento_984d1b9.pdf) - this adds a cover to the previous document. It uses OpenPDF for it.
They are the same public document. I only can provide this Link with captcha - ( https://pjedes.trt12.jus.br/consultaprocessual/detalhe-processo/0000774-13.2022.5.12.0030/1#984d1b9 )
2. With NVDA, search for this text (use Ctrl+<NVDA key>+F): Ainda, tendo em vista o disposto
3. try to keep reading with NVDA (just use arrow down for each line).  

What is the expected behavior? (add screenshot)
We expect to NVDA to be able to read the document in the same way we see it.
![image](https://github.com/mozilla/pdf.js/assets/34720010/2890d633-f049-41cf-b5dc-f5a6f404a8f7)

What went wrong? (add screenshot)

Here it is how NVDA reads the file (I highlighted the wrong parts)

 Ainda, tendo em vista o disposto no § 
1º ao art. 840 da CLT, oqual estabelece a necessidade de "pedido certo, determinado e com a 
indicação de seu", bem como a previsão contida no art. 492 do CPC ("**valor**é vedado ao juiz 
proferirdecisão de natureza diversa da pedida, bem como condenar a parte em quantidade**)"**, superior 
ou em objeto diverso do que lhe foi demandadodeverão ser observadospara efeito de limitação de 
eventual condenação os valores referentes aos pedidos**com  exceção  dos  juros  e  da  correção**
indicados  pela  parte  autora  na  exordial,monetária. Cerceamento de defesa - 

It is very hard to read for those who need to use NVDA. This is just an paragraph.

I tried to open these files offline in Firefox and Edge (chromium):
Edge reads it correctly;. Firefox still have problems reading them

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension): As 

Our system uses embedded pdf.js, but the problem is the same using pdf.js hosted by mozilla.

I am trying to improve the system accessibility wise, but I do not know how to help with pdf.js (they complain a lot (they are right!), they have to download every file to open it in adobe acrobat)

Thanks in advance. 
A. Lucas

**Comment by antoniolucasnobar** [comment]
can anyone point directions to what part of pdf.js investigate?

**Comment by joanlambert** [comment]
Our team is having the same problem, identified with JAWS--PDF.js does not honor spaces at the end of a line break within a paragraph, so the last word in each line runs into the first word of the next. It's a terrible experience for a screen reader user.
